src/test/scala/com/crealytics/spark/v2/excel/WriteAndReadSuite.scala

```diff
@@ -126,4 +155,42 @@ class WriteAndReadSuite extends AnyFunSuite with DataFrameSuiteBase with ExcelTe
     })
   }
 
+  test("write then read java.sql.date and java.sql.timestamp") {
+    val path = Files.createTempDirectory("spark_excel_wr_03_").toString()
+    spark.conf.set(DATETIME_JAVA8API_ENABLED, false)
+    val df_source = spark.createDataFrame(expectedData_03_sql, userDefinedSchema_03).sort("Id")
+    df_source.write.format("excel").mode(SaveMode.Append).save(path)
+
+    val df_read = spark.read
+      .format("excel")
+      .schema(userDefinedSchema_03)
+      .load(path)
+      .sort("Id")
+
+    assertDataFrameEquals(df_source, df_read)
+
+    /* Cleanup, should after the checking */
+    spark.conf.unset(DATETIME_JAVA8API_ENABLED)
```

Comment:
Can we `.get` the value before and then `.set` to that? With the current implementation, we would be losing the previous value.

Reply:
done